### PR TITLE
feat: add useBreakpoint hook to determine device type

### LINKED ISSUE #361: Create `useBreakpoint` hook
### Package / App

MAD Components

### Description

Similar check as is used in App.tsx for Chronicles, except we should have componets export the hook instead (since it is style related). Ideally, the hook should respond with sizes as strings ("sm", "lg" etc. etc.) so that users don't have to care about the specific pixel sizes. See [tailwind breakpoints](https://tailwindcss.com/docs/responsive-design) for some good breakpoint values.

### Alternatives

_No response_

### Other info

Mock implementation:

```ts
const { width } = useWindowDimensions();
    return useMemo(() => {
        return width > 576 ? "sm" : "md";
        // mer breakpoints??
    }, [width]);

```

### PR DESCRIPTION
**Pull request by Sandtory**
add `useBreakpoint` hook and `Breakpoint` type to package `mad-components` 

closes #361

### CODE REVIEW

File: apps/chronicles/App.tsx [hunk 1]
```diff
@@ -2,22 +2,22 @@ import React, { useEffect, useMemo } from "react";
...
     const isLoadingComplete = useCachedResources();
     const [hasLoadedEds] = useEDS();
     const colorScheme = useColorScheme();
 
-    const { width } = useWindowDimensions();
+    const breakpoint = useBreakpoint();
     const deviceType = useMemo(() => {
-        return width > 576 ? "tablet" : "phone";
-    }, [width]);
+        console.log(breakpoint);
```
Comment (HaakonSvane) [inline apps/chronicles/App.tsx:18]: Console log here

File: .changeset/olive-sheep-run.md [hunk 2]
```diff
@@ -0,0 +1,6 @@
+---
+"@equinor/mad-components": patch
+---
+
+Add useBreakpoint hook to use for apps and components that have specific requirements or styles
```
Comment (HaakonSvane) [inline .changeset/olive-sheep-run.md:5]: I think changelogs should be written in past tense. For eksempel: Added useBreakpoint hook...

**Review by HaakonSvane — CHANGES_REQUESTED** [review]
😡 💯

**Comment by HaakonSvane** [comment]
Remember to reference which issue this PR closes in the PR description

File: apps/chronicles/App.tsx (hunk 1 above)
Comment (Sandtory) [reply apps/chronicles/App.tsx:18]: no console log anymore 😡

File: .changeset/olive-sheep-run.md (hunk 2 above)
Comment (Sandtory) [reply .changeset/olive-sheep-run.md:5]: not present tense anymore, changed to past tense

**Review by HaakonSvane — APPROVED** [review]
👍 😠 🥇 👿

**Pull request merged**